SteamKit2/SteamKit2/Steam/Handlers/SteamFriends/Callbacks.cs

```diff
@@ -493,6 +493,40 @@ internal FriendMsgEchoCallback( CMsgClientFriendMsgIncoming msg )
         }
 
         /// <summary>
+        /// This callback is fired in response to receiving historical messages. 
+        /// This can be due to a call to <see cref="RequestOfflineMessages"/>
+        /// or due to a call to <see cref="RequestMessageHistory(SteamID)"/>.
+        /// </summary>
+        public sealed class FriendMsgHistoryCallback : CallbackMsg
+        {
+            /// <summary>
+            /// Gets the result of the request.
+            /// </summary>
+            public EResult Result { get; private set; }
+
+            /// <summary>
+            /// Gets the SteamID of the user with whom these messages were exchanged.
+            /// </summary>
+            public SteamID SteamID { get; private set; }
+
+            /// <summary>
+            /// The messages exchanged with the user.
+            /// Offline messages are marked by having set 
+            /// <see cref="CMsgClientFSGetFriendMessageHistoryResponse.FriendMessage.unread"/> to true
+            /// </summary>
+            public List<CMsgClientFSGetFriendMessageHistoryResponse.FriendMessage> Messages { get; private set; }
```

Comment:
We'd want to wrap this into an inner class, check out how the `SteamApps` callbacks do this.